File: OmniCore.Py/exceptions.cs
namespace OmniCore.Py
{
    using System;

    public class OmnipyError : Exception
    {
        public OmnipyError(string message = "Unknown", Exception inner = null) : base(message, inner) { }
    }

    public class PacketRadioError : OmnipyError
    {
        public PacketRadioError(string message = "Unknown RL error", Exception innerException = null) : base(message, innerException)
        {
        }
    }

    public class ProtocolError : OmnipyError
    {
        public RadioPacket ReceivedPacket = null;
        public ProtocolError(string message = "Unknown protocol error", RadioPacket received = null) : base(message)
        {
            this.ReceivedPacket = received;
        }
    }

    public class OmnipyTimeoutError : OmnipyError
    {
        public OmnipyTimeoutError(string message = "Timeout error") : base(message) { }
    }

    public class PdmError : OmnipyError
    {
        public PdmError(string message = "Unknown pdm error", Exception innerException = null) : base(message, innerException)
        {
        }
    }

    public class PdmBusyError : PdmError
    {
        public PdmBusyError(string message = "Pdm is busy.") : base(message) { }
    }
}

File: OmniCore.Py/pyref/exceptions.py

class OmnipyError(Exception):
    def __init__(self, message="Unknown"):
        self.error_message = message


class PacketRadioError(OmnipyError):
    def __init__(self, message="Unknown RL error", err_code=None):
        OmnipyError.__init__(self, message)
        self.err_code = err_code


class ProtocolError(OmnipyError):
    def __init__(self, message="Unknown protocol error"):
        OmnipyError.__init__(self, message)

class OmnipyTimeoutError(OmnipyError):
    def __init__(self, message="Timeout error"):
        OmnipyError.__init__(self, message)

class PdmError(OmnipyError):
    def __init__(self, message="Unknown pdm error"):
        OmnipyError.__init__(self, message)


class PdmBusyError(PdmError):
    def __init__(self, message="Pdm is busy."):
        PdmError.__init__(self, message)
